actions/transactions/broadcast_callback.go

```diff
@@ -12,21 +12,18 @@ import (
 // broadcastCallback will handle a broadcastCallback call from the broadcast api
 func broadcastCallback(c *gin.Context) {
 	logger := reqctx.Logger(c)
-	var resp *broadcast.SubmittedTx
+	var callbackResp *broadcast.SubmittedTx
 
-	err := c.Bind(&resp)
+	err := c.Bind(&callbackResp)
 	if err != nil {
 		spverrors.ErrorResponse(c, spverrors.ErrCannotBindRequest, logger)
 		return
 	}
 
-	err = reqctx.Engine(c).UpdateTransaction(c.Request.Context(), resp)
+	err = reqctx.Engine(c).HandleTxCallback(c.Request.Context(), callbackResp)
 	if err != nil {
-		logger.Err(err).Msgf("failed to update transaction - tx: %v", resp)
-		spverrors.ErrorResponse(c, err, logger)
-		return
+		logger.Err(err).Msgf("failed to update transaction - tx: %v", callbackResp)
```

Comment:
why here is different way of logging not like above in line 19 `spverrors.ErrorResponse(c, err, logger)`

Reply:
Good point 👍 But it seems to be ok, because:

Generally as @wregulski said, the ARC doesn't need to know about our internal errors.
It needs only one information - do we received the "notifiaction" or not. 

In the line `19` there is a 400 error - means that the ARC sent us wrong data in json - so it is their fault - that's why we return `status.400` and (as a side effect we log the error).

However, in the line `25` there is handling of failed processing on our side.  Means, the ARC successfully notified us about the transaction but we somehow didn't parse it well. 
- In this case we return `status.200` (you ARC sent the notification correctly) and only log a message for internal (e.g. for debugging purposes - why that happened)